quarkus/config-api/src/main/java/org/keycloak/config/OptionBuilder.java

```diff
@@ -93,13 +94,24 @@ public OptionBuilder<T> expectedValues(List<String> expected) {
     }
 
     public OptionBuilder<T> expectedValues(Class<? extends Enum> expected) {
-        return expectedValues(Stream.of(expected.getEnumConstants()).map(Object::toString).collect(Collectors.toList()));
+        return expectedValues(Stream.of(expected.getEnumConstants())
+                .map(Object::toString)
+                .map(value -> transformEnumValues ? Option.transformEnumValue(value) : value)
```

Comment:
Is there a way to do this where there isn't a inter-dependency between expectedValues and transformEnumValues? If you set transformEnumValues after calling expectedValues for example you get different behavior.

Reply:
Nice point, updated.